app/jobs/create_story_excerpt_job.rb

```diff
@@ -29,7 +29,7 @@ def create_excerpt_content(story)
 
     @client.completions(
       parameters: {
-        model: 'text-davinci-003',
```

Comment:
The completions endpoint does not support this model. Please refer to the gem documentation to learn how to use chat completions.

Using this model with the completions endpoint responds:

```ruby
{"error"=>
  {"message"=>"This is a chat model and not supported in the v1/completions endpoint. Did you mean to use v1/chat/completions?",
   "type"=>"invalid_request_error",
   "param"=>"model",
   "code"=>nil}}
   ```
   
   It looks like you just changed the code and you didn't test if it worked.

Reply:
That's interesting! In the `app/jobs/create_story_excerpt_job.rb` file we have this code:
```
    @client.completions(
      parameters: {
        model: 'gpt-3.5-turbo',
```
Why is the error showing the name of the old model?

There's one more thing, when I run `rails db:create` it shows me this error:
```
KeyError: key not found: "ACTIVE_RECORD_ENCRYPTION_PRIMARY_KEY"
```
I have also noticed that there are no credentials given to create the database. I am curious, how that part is being handled?

Reply:
Regarding the error, it's very clear, you are passing a chat model to a completions endpoint rather than a chat/completions endpoint.

Regarding the `rails db:create`

 Did you run the commands listed in the readme?

Specifically

```
Run bin/rails db:encryption:init and copy each key to the .env file
```

Reply:
### Update: 
So I actually used the name of the ChatGPT model `gpt-3.5-turbo` instead of the ChatGPT Completion model `text-davinci-001`.

Reply:
Yes

> Regarding the error, it's very clear, you are passing a chat model to a completions endpoint rather than a chat/completions endpoint.
> 
> Regarding the `rails db:create`
> 
> Did you run the commands listed in the readme?
> 
> Specifically
> 
> ```
> Run bin/rails db:encryption:init and copy each key to the .env file
> ```
Yes! I actually googled the error and one of the StackOverflow answers suggested running this command first! But I received the same error message!

Reply:
> Yes! I actually googled the error and one of the StackOverflow answers suggested running this command first! But I received the same error message!

I believe you didn't save them in the .env then, you just ran the command.

Reply:
> Sorry for the novice mistake there! I have never used this gem and I am only starting to explore it all thanks to you for giving me this opportunity!

No worries, I'm glad to teach and help.

Reply:
> 
> But the problem is you are using the wrong endpoint
> 
> `@client.completions(`
> 
> instead of
> 
> `@client.chat`
> 
So I believe you want me to use `@client.chat` instead of `@clien.completion`. And `completion` uses `GPT-3` model which is older I guess or at least we aren't using it here. So I need to replace that!

Reply:
Yes!

Reply:
```
    response = client.chat(
        parameters: {
            model: "gpt-3.5-turbo",
            messages: [{ role: "user", content: "Hello!"}],
        })
```
With the newer model, they haven't mentioned the `max_tokens` key inside the `parameters`. I have read the resources that you provided but still, I am not able to understand where I should be defining this. Are these set automatically? I highly doubt that!

As for the `messages` param is concerned, I believe you want to use `role` as a `user` and `content` with `prompt` & you don't want to use other roles such as `system` or `assistant`, right?

Reply:
Here they mention the max_tokens

https://platform.openai.com/docs/api-reference/chat/create

For now, just use user role

Reply:
![Screenshot from 2023-03-02 23-32-55](https://user-images.githubusercontent.com/10782481/222513338-11c1582b-679c-4743-a035-4b02fbf9573a.png)

Successfully able to generate images and text without any errors with the new model, except for this warning which I believe is local :
```
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:51: warning: already initialized constant GLib::G_FREE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:51: warning: previous definition of G_FREE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:59: warning: already initialized constant GLib::LOG_FLAG_RECURSION
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:59: warning: previous definition of LOG_FLAG_RECURSION was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:60: warning: already initialized constant GLib::LOG_FLAG_FATAL
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:60: warning: previous definition of LOG_FLAG_FATAL was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:63: warning: already initialized constant GLib::LOG_LEVEL_ERROR
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:63: warning: previous definition of LOG_LEVEL_ERROR was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:64: warning: already initialized constant GLib::LOG_LEVEL_CRITICAL
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:64: warning: previous definition of LOG_LEVEL_CRITICAL was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:65: warning: already initialized constant GLib::LOG_LEVEL_WARNING
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:65: warning: previous definition of LOG_LEVEL_WARNING was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:66: warning: already initialized constant GLib::LOG_LEVEL_MESSAGE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:66: warning: previous definition of LOG_LEVEL_MESSAGE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:67: warning: already initialized constant GLib::LOG_LEVEL_INFO
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:67: warning: previous definition of LOG_LEVEL_INFO was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:68: warning: already initialized constant GLib::LOG_LEVEL_DEBUG
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:68: warning: previous definition of LOG_LEVEL_DEBUG was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:71: warning: already initialized constant GLib::GLIB_TO_SEVERITY
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:71: warning: previous definition of GLIB_TO_SEVERITY was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:86: warning: already initialized constant GLib::LOG_HANDLER
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:86: warning: previous definition of LOG_HANDLER was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:155: warning: already initialized constant GObject::GBOOL_TYPE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:155: warning: previous definition of GBOOL_TYPE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:156: warning: already initialized constant GObject::GINT_TYPE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:156: warning: previous definition of GINT_TYPE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:157: warning: already initialized constant GObject::GUINT64_TYPE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:157: warning: previous definition of GUINT64_TYPE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:158: warning: already initialized constant GObject::GDOUBLE_TYPE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:158: warning: previous definition of GDOUBLE_TYPE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:159: warning: already initialized constant GObject::GENUM_TYPE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:159: warning: previous definition of GENUM_TYPE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:160: warning: already initialized constant GObject::GFLAGS_TYPE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:160: warning: previous definition of GFLAGS_TYPE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:161: warning: already initialized constant GObject::GSTR_TYPE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:161: warning: previous definition of GSTR_TYPE was here
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:162: warning: already initialized constant GObject::GOBJECT_TYPE
/usr/share/rvm/gems/ruby-3.2.0/gems/ruby-vips-2.1.4/lib/vips.rb:162: warning: previous definition of GOBJECT_TYPE was here
[ActiveJob] [ActiveStorage::AnalyzeJob] [f7231a56-a722-4d42-b871-c1812eb0bdb2] Skipping image analysis because the ruby-vips gem isn't installed

```

I have created multiple stories and this warning triggers whenever a new image is being generated. So I have made multiple changes across files, do you want me to commit all the changes in one and create a new PR ?

Reply:
Yes it looks a problem with your vips gem, no worries.

Please push changes in this PR.